honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,364 @@
+        else:
+            for k in range(len(mesh.VertexColors)):
+                col = mesh.VertexColors[k]
+                colors.append(lbc.Color(col.R, col.G, col.B))
+    return lb_faces, colors
+
+
+def mesh_to_mesh3d(mesh, color_by_face=True):
+    """Get a Ladybug Mesh3D object from a Rhino3dm Mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A Ladybug Mesh3D object
+    """
+    lb_verts = tuple(to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices)))
+    lb_faces, colors = extract_mesh_faces_colors(mesh, color_by_face)
+    return Mesh3D(lb_verts, lb_faces, colors)
+
+
+def mesh_to_face3d(mesh):
+    """Get a list of Ladybug Face3D objects from a rhino3dm Mesh.
+
+    Args:
+        mesh: A rhino3dm mesh geometry.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+
+    faces = []
+    pts = [to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices))]
+
+    for j in range(len(mesh.Faces)):
+        face = mesh.Faces[j]
+        if len(face) == 4:
+            all_verts = (pts[face[0]], pts[face[1]],
+                         pts[face[2]], pts[face[3]])
+        else:
+            all_verts = (pts[face[0]], pts[face[1]], pts[face[2]])
+
+        faces.append(Face3D(all_verts))
+
+    return faces
+
+
+def brep_to_meshed_face3d(brep):
+    """Get a Ladybug Face3D object from a planar rhino3dm Brep.
+
+    This method meshes the brep in order to create a Ladybug
+    Face3D object.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A Ladybug Face3D object.
+    """
+
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+    polyface = Polyface3D.from_faces(faces, 0.01)
+    lines = list(polyface.naked_edges)
+    polylines = Polyline3D.join_segments(lines, 0.01)
+    face3d = Face3D(boundary=polylines[0].vertices)
+
+    return face3d
+
+
+def brep_to_mesh_to_face3d(brep):
+    """Get a list of Ladybug Face3D objects from a rhino3dm Brep.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+    return faces
+
+
+def brep_to_face3d(brep, tolerance):
+    """Get a list of Ladybug Face3D objects from a planar rhino3dm Brep.
+
+    This function is used in translating rhino3dm planar Brep to Ladybug Face3D objects.
+
+    Args:
+        brep: A rhino3dm Brep.
+        tolerance: A rhino3dm tolerance object. Tolerance set in the rhino file.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    # Getting all vertices of the face
+    mesh = brep.Faces[0].GetMesh(rhino3dm.MeshType.Any)
```

Comment:
Does this mean that Brep geometry must always be exploded in order to use your library here?  It's just a suggestion but maybe you should just iterate through the faces and translate each one to a `Face3D`.

Reply:
Yes, in this library, we always create a Face3D for each of the face of a rhino3dm Brep. I am already doing that. I am guessing that this suggestion stems from you observation that I am only using one face of the brep here. `brep.Faces[0]`. PLease note that this method is only employed for planar Breps, so in most cases there's only one face. For the rest of the cases, I run the Brep through the `brep_to_meshed_face3d` method where the brep is converted into a mesh > The mesh faces are converted into Face3ds > A plant polysurface is created from face3ds > A combined Face3d is created from the polyface3d.